qiskit/utils/quantum_instance.py

```diff
@@ -967,22 +974,31 @@ def backend(self):
     @property
     def backend_name(self):
         """Return backend name."""
-        return self._backend.name()
+        if self._backend_version <= 1:
+            return self._backend.name()
+        else:
+            return self._backend.name
 
     @property
     def is_statevector(self):
         """Return True if backend is a statevector-type simulator."""
-        return is_statevector_backend(self._backend)
+        if self._backend_version <= 1:
+            return is_statevector_backend(self._backend)
+        return False
 
     @property
     def is_simulator(self):
         """Return True if backend is a simulator."""
-        return is_simulator_backend(self._backend)
+        if self._backend_version <= 1:
+            return is_simulator_backend(self._backend)
+        return False
```

Comment:
I don't think this is necessary. (See next comment)

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7563/commits/bfdf394f4e4e3fdf9314f6b402dbdccae238b656